equinox/nn/_embedding.py

```diff
@@ -176,19 +181,17 @@ def rotate_half(x: Float[Array, "seq_length embedding_size"]):
 
     @staticmethod
     def precompute_freqs_cis(
-        embedding_size: int, end: int, theta: float
-    ) -> Complex[Array, "end half_of_embedding_size"]:
+        embedding_size: int, end: int, theta: float, dtype: Any
+    ) -> tuple[Float[Array, "end half_emb_size"], Float[Array, "end half_emb_size"]]:
         freqs = 1.0 / (
             theta
             ** (jnp.arange(0.0, embedding_size, 2)[jnp.newaxis, :] / embedding_size)
         )
 
         t = jnp.arange(float(end))
         freqs_outer = jnp.outer(t, freqs)
-        with jax.numpy_dtype_promotion("standard"):
-            freqs_cis = jnp.cos(freqs_outer) + jnp.sin(freqs_outer) * 1j
 
-        return freqs_cis
+        return jnp.cos(freqs_outer).astype(dtype), jnp.sin(freqs_outer).astype(dtype)
```

Comment:
Might be worth including a comment here, to emphasise that the usual rule of "prefer `some_constructor(..., dtype=...)` over `.astype(dtype)`" does not apply.